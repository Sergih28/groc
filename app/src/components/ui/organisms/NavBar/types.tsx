export type NavBarElement = {
  name: string
  url: string
  isEnabled: boolean
}
