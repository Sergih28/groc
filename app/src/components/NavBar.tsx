type NavBarElement = {
  name: string
  url: string
  isEnabled: boolean
}

const NAVBAR_STYLES = {
  LIST: 'flex flex-col gap-4',
  DISABLED_ELEMENT: 'cursor-not-allowed text-gray-400',
  ENABLED_ELEMENT: 'cursor-pointer text-blue-500 hover:text-blue-700',
}

export const NAVBAR_ELEMENTS: NavBarElement[] = [
  {
    name: 'Home',
    url: '/',
    isEnabled: true,
  },
  {
    name: 'Pomodoro',
    url: '/pomodoro',
    isEnabled: false,
  },
  {
    name: 'Notifications',
    url: '/notifications',
    isEnabled: false,
  },
  {
    name: 'Settings',
    url: '/settings',
    isEnabled: false,
  },
  {
    name: 'Stats',
    url: '/stats',
    isEnabled: false,
  },
  {
    name: 'Challenges',
    url: '/challenges',
    isEnabled: false,
  },
  {
    name: 'Export',
    url: '/export',
    isEnabled: false,
  },
  {
    name: 'Help',
    url: '/help',
    isEnabled: false,
  },
]

const NavBar = () => {
  return (
    <ul className={NAVBAR_STYLES.LIST}>
      {NAVBAR_ELEMENTS.map(({ url, name, isEnabled }) => {
        const ListElement = () => (
          <li
            className={isEnabled ? NAVBAR_STYLES.ENABLED_ELEMENT : NAVBAR_STYLES.DISABLED_ELEMENT}
          >
            {name}
          </li>
        )

        if (!isEnabled) return <ListElement />

        return (
          <a href={url}>
            <ListElement />
          </a>
        )
      })}
    </ul>
  )
}

export default NavBar
