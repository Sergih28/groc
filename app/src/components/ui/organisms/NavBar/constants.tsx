import type { NavBarElement } from './types'

export const NAVBAR_ELEMENTS: NavBarElement[] = [
  {
    name: 'Home',
    url: '/',
    isEnabled: true,
  },
  {
    name: 'Pomodoro',
    url: '/pomodoro',
    isEnabled: false,
  },
  {
    name: 'Notifications',
    url: '/notifications',
    isEnabled: false,
  },
  {
    name: 'Settings',
    url: '/settings',
    isEnabled: false,
  },
  {
    name: 'Stats',
    url: '/stats',
    isEnabled: false,
  },
  {
    name: 'Challenges',
    url: '/challenges',
    isEnabled: false,
  },
  {
    name: 'Export',
    url: '/export',
    isEnabled: false,
  },
  {
    name: 'Help',
    url: '/help',
    isEnabled: false,
  },
] as const
