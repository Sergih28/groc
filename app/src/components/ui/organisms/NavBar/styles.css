.navbar__list {
  @apply m-2 flex flex-col gap-5 text-xl;
}

.navbar__disabled-element {
  @apply cursor-not-allowed text-gray-400;
}

.navbar__enabled-element {
  @apply cursor-pointer text-blue-500 hover:text-blue-700;
}
