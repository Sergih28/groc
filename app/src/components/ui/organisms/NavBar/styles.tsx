export const STYLES = {
  LIST: 'flex flex-col gap-4',
  DISABLED_ELEMENT: 'cursor-not-allowed text-gray-400',
  ENABLED_ELEMENT: 'cursor-pointer text-blue-500 hover:text-blue-700',
} as const
