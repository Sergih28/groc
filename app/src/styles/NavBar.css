#navbar {
  @apply row-span-5 bg-slate-900;
}
