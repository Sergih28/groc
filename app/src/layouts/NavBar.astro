---
import ThemeToggle from '@molecules/ThemeToggle'
import NavBarComponent from '@organisms/NavBar'

import '@styles/NavBar.css'
---

<aside id="navbar">
  <div
    class="m-auto mb-10 mt-2 flex h-[100px] w-full items-center justify-center bg-yellow-300 text-center text-3xl font-bold text-white"
  >
    <span>GROC</span>
  </div>
  <NavBarComponent client:only="react" />
  <ThemeToggle client:only="react" />
</aside>
