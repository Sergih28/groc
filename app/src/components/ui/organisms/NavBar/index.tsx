import { NAVBAR_ELEMENTS } from './constants'
import { type ValUIType } from '@i18n/ui'
import { getLangFromWindowUrl, useTranslations } from '@i18n/utils'

import { STYLES } from './styles'

const NavBar = () => {
  const lang = getLangFromWindowUrl()
  const t = useTranslations(lang)

  return (
    <ul className={STYLES.LIST}>
      {NAVBAR_ELEMENTS.map(({ url, key, isEnabled }) => {
        const ListElement = () => (
          <li className={isEnabled ? STYLES.ENABLED_ELEMENT : STYLES.DISABLED_ELEMENT}>
            {t(`nav.${key}` as ValUIType)}
          </li>
        )

        if (!isEnabled) return <ListElement key={key} />

        return (
          <a key={key} href={`/${lang}${url}`}>
            <ListElement />
          </a>
        )
      })}
    </ul>
  )
}

export default NavBar
