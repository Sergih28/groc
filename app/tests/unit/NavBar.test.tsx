import NavBar, { NAVBAR_ELEMENTS } from '@components/NavBar'
import { render } from '@testing-library/react'

describe('NavBar component', () => {
  test('renders the NavBar component', () => {
    render(<NavBar />)
  })

  test('NavBar elements snapshot', () => {
    expect(NAVBAR_ELEMENTS).toMatchSnapshot()
  })
})
