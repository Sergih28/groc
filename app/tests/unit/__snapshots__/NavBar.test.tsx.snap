// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar component > NavBar elements snapshot 1`] = `
[
  {
    "isEnabled": true,
    "name": "Home",
    "url": "/",
  },
  {
    "isEnabled": false,
    "name": "Pomodoro",
    "url": "/pomodoro",
  },
  {
    "isEnabled": false,
    "name": "Notifications",
    "url": "/notifications",
  },
  {
    "isEnabled": false,
    "name": "Settings",
    "url": "/settings",
  },
  {
    "isEnabled": false,
    "name": "Stats",
    "url": "/stats",
  },
  {
    "isEnabled": false,
    "name": "Challenges",
    "url": "/challenges",
  },
  {
    "isEnabled": false,
    "name": "Export",
    "url": "/export",
  },
  {
    "isEnabled": false,
    "name": "Help",
    "url": "/help",
  },
]
`;
